fix: Make the engine destroyable before the loop ends.

ThomasRuijters opened the pull request:
Due to a goof with static memory and singletons not being nice, we implemented a workaround that should manage the singleton from the instance, not the static class alone.

ThomasRuijters committed:
fix: Make the engine destroyable before the loop ends.

Due to a goof with static memory and singletons not being nice, we implemented a workaround that should manage the singleton from the instance, not the static class alone.

BotSquirrel commented on include/engine/core/engine.h:
```diff
@@ -1,16 +1,22 @@
 #pragma once
+
 #include <memory>
+#include <iostream>
```
`<iostream>` is an artefact from debugging and could be removed but I am not blocking over just this

SjulGr commented on include/engine/core/engine.h:
```diff
@@ -1,16 +1,22 @@
+
     Engine();
+
+    static std::unique_ptr<Engine, Deleter> engine_instance;
```
Coding guidelines; ending private members on an underscore (`engine_instance` -> `engine_instance_`)

ThomasRuijters committed:
Remove unused iostream include from engine.h

ThomasRuijters committed:
Rename engine_instance to engine_instance_

ThomasRuijters committed:
Rename engine_instance to engine_instance_

ThomasRuijters committed:
fix: Linter:

... do I really have to...

ThomasRuijters committed:
fix: Suppress the linter on the static member.

So, clang-tidy has a skill issue in recognising static members as static members, and it thinks they are variables. Not good. So the suppression will have to do for now. Not pretty, but this is an internal problem inside clang-tidy that we can not touch. The config is set to deal with  static members properly. The linter just does not see our static member variables as statics but rather as normal variables...
